springfox-spring-web/src/main/java/springfox/documentation/spring/web/readers/operation/ModelRefs.java

```diff
@@ -58,7 +61,8 @@ public static ModelRef modelRef(ResolvedType resolved,
     if (Void.class.equals(resolved.getErasedType()) || Void.TYPE.equals(resolved.getErasedType())) {
       return new ModelRef("void");
     }
+    AllowableValues allowableValues = ResolvedTypes.allowableValues(resolved);
     String typeName = nameExtractor.typeName(fromParent(modelContext, resolved));
-    return new ModelRef(typeName);
+    return new ModelRef(typeName, null, allowableValues);
```

Comment:
@dilipkrish I'm hesitant to add a builder at this point because then I'd feel compelled to change all references of `_new Model(...)_ to use the builder.  Perhaps this could be done as part of your planned refactoring?

I'll add a separate constructor to eliminate the null for now.  Will that work?